#834 #256 fix for unable to load the Indexes page

utajum opened the pull request:
#834 #256 fix for unable to load the Indexes page

#### Database Migration
NO

#### Description
Do not unload the DOM for a simple undefined property error.
Tested it with the Prowlarr backend and it works great.

#### Screenshot (if UI related)
Fix for this ![](https://user-images.githubusercontent.com/4707492/122823051-c717eb80-d2d6-11eb-9305-e5f5bcb387d7.png)


#### Todos
- [ ] Tests
- [ ] Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)
- [ ] [Wiki Updates](https://wiki.servarr.com)

#### Issues Fixed or Closed by this PR

* Fixes #834 #256

Qstick commented:
An indexer should never be without app profile, so there are likely other issues that need addressing here

utajum commented:
IMHO yes the problem is somewhere else, but that should not break the UI. A broken DB state should not unload the DOM. 
It prevents the user from removing a faulty DB record, which brakes the user experience.

For me it broke at random while adding the `legittorrents` indexer, somehow the appProfile record in my db was set to 0, changed it manually to 1 and the problem went away.
I also removed that specific indexer (from prowlarr UI), then added it again and it had no problems. 🤷‍♂
